src/main/java/com/google/javascript/gents/TypeConversionPass.java

```diff
@@ -133,44 +136,61 @@ public void visit(NodeTraversal t, Node n, Node parent) {
    * Converts fields declared internally inside a class using the "this" keyword.
    */
   private class FieldOnThisConverter extends AbstractPostOrderCallback {
+    // Map from class node to its field names
+    private Multimap<Node, String> classFieldMap = HashMultimap.create();
+
     @Override
     public void visit(NodeTraversal t, Node n, Node parent) {
       if (n.isExprResult()) {
         ClassMemberDeclaration declaration = ClassMemberDeclaration.newDeclarationOnThis(n);
 
-        if (declaration == null) {
+        if (declaration == null || declarationHasBeenAdded(declaration)) {
           return;
         }
 
-        // TODO(gmoothart): in many cases we should be able to infer the type from the rhs
-        if (declaration.jsDoc != null &&
-            declaration.jsDoc.getType() != null) {
-          Node fnNode = NodeUtil.getEnclosingFunction(n);
-          String fnName = fnNode.getParent().getString();
+        Node fnNode = NodeUtil.getEnclosingFunction(n);
+        String fnName = fnNode.getParent().getString();
+
+        // TODO(gmoothart): in many cases we should be able to infer the type from the rhs if there
+        // is no jsDoc
+
+        // Handle parameter properties when we are in the constructor and have a
+        // declaration of the form this.name = name;
```

Comment:
add how you handle them? I.e. "by converting them to private/public constructor parameters" or something like that?